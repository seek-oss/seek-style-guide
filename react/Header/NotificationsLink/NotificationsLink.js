import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import BellIcon from '../../BellIcon/BellIcon';
import styles from './NotificationsLink.less';

const NotificationsLink = ({
  isAuthenticated,
  linkRenderer,
  isInExperiment
}) => {
  return isInExperiment && isAuthenticated ? (
    <Fragment>
      {linkRenderer({
        href: '/notifications',
        children: <BellIcon className={styles.bell} />
      })}
      <span className={styles.bellDivider} />
    </Fragment>
  ) : null;
};

NotificationsLink.propTypes = {
  isAuthenticated: PropTypes.bool,
  linkRenderer: PropTypes.func,
  isInExperiment: PropTypes.bool
};

export default NotificationsLink;
