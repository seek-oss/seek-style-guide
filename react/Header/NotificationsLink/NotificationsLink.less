@import (reference) "~seek-style-guide/theme";
@import (reference) "../Header.less";

.bell {
  padding: 15px 0;
  position: relative;
  top: 1px;
  color: white;

  @media @desktop {
    color: @sk-black;
    display: inline-block;
  }
}

.bellDivider {
  .divider();
  display: inline-block;

  @media @mobile {
    margin-right: 0;
    border-right: @divider-width solid white;
    opacity: 0.2;
  }
}
