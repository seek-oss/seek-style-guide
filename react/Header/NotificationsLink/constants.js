export const EXPERIMENT_ID = '58CC4E12-A4E0-4C0B-BD2A-681FD98FDC2E';
export const AB_CRUNCH_URL =
  'https://experiments.cloud.seek.com.au/participants';
