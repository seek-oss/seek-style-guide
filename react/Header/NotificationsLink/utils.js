export function getCookieFromString(key, cookieString) {
  const value = `; ${cookieString}`;
  const parts = value.split(`; ${key}=`);
  if (parts.length === 2) {
    return parts
      .pop()
      .split(';')
      .shift();
  }
  return null;
}
