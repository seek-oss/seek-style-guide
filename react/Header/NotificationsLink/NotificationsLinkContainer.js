import React from 'react';
import PropTypes from 'prop-types';
import { getCookieFromString } from './utils';
import {
  AUTHENTICATED,
  UNAUTHENTICATED,
  AUTH_PENDING
} from '../../private/authStatusTypes';
import { EXPERIMENT_ID, AB_CRUNCH_URL } from './constants';
import NotificationsLink from './NotificationsLink';

class NotificationsLinkContainer extends React.Component {
  static propTypes = {
    authenticationStatus: PropTypes.oneOf([
      AUTHENTICATED,
      UNAUTHENTICATED,
      AUTH_PENDING
    ]),
    linkRenderer: PropTypes.func
  };

  state = {
    isInExperiment: false
  };

  componentDidMount() {
    let visitorId = null;

    try {
      if (typeof document !== 'undefined') {
        visitorId = getCookieFromString('JobseekerVisitorId', document.cookie);
      }
      if (visitorId) {
        const url = `${AB_CRUNCH_URL}/${visitorId}`;

        fetch(url)
          .then(response => response.json())
          .then(response => {
            const isInExperiment = Boolean(response.experiments[EXPERIMENT_ID]);

            if (isInExperiment) {
              this.setState({ isInExperiment: true });
            }
          })
          .catch(() => {});
      }
      // eslint-disable-next-line no-empty
    } catch (e) {}
  }

  render() {
    const { authenticationStatus, linkRenderer } = this.props;
    const isAuthenticated = authenticationStatus === AUTHENTICATED;

    return (
      <NotificationsLink
        isAuthenticated={isAuthenticated}
        isInExperiment={this.state.isInExperiment}
        linkRenderer={linkRenderer}
      />
    );
  }
}

export default NotificationsLinkContainer;
