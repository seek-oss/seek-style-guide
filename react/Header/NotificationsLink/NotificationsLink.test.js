import React from 'react';
import { shallow } from 'enzyme';
import NotificationsLink from './NotificationsLink';

const renderNotificationsLink = props =>
  shallow(<NotificationsLink linkRenderer={() => <a />} {...props} />);

describe('NotificationsLink:', () => {
  describe('when the user is not authenticated', () => {
    it('and they are inExperiment, they should not see the link', () => {
      const component = renderNotificationsLink({
        isAuthenticated: false,
        isInExperiment: true
      });
      expect(component).toMatchSnapshot();
    });

    it('and they are not inExperiment, they should not see the link', () => {
      const component = renderNotificationsLink({
        isAuthenticated: false,
        isInExperiment: false
      });
      expect(component).toMatchSnapshot();
    });
  });

  describe('when the user is authenticated', () => {
    it('and they are inExperiment, they should see the link', () => {
      const component = renderNotificationsLink({
        isAuthenticated: true,
        isInExperiment: true
      });
      expect(component).toMatchSnapshot();
    });

    it('and they are not inExperiment, they should not see the link', () => {
      const component = renderNotificationsLink({
        isAuthenticated: true,
        isInExperiment: false
      });
      expect(component).toMatchSnapshot();
    });
  });
});
