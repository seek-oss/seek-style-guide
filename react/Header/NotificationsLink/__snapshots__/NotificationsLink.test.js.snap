// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationsLink: when the user is authenticated and they are inExperiment, they should see the link 1`] = `
<Fragment>
  <a />
  <span
    className="NotificationsLink__bellDivider"
  />
</Fragment>
`;

exports[`NotificationsLink: when the user is authenticated and they are not inExperiment, they should not see the link 1`] = `""`;

exports[`NotificationsLink: when the user is not authenticated and they are inExperiment, they should not see the link 1`] = `""`;

exports[`NotificationsLink: when the user is not authenticated and they are not inExperiment, they should not see the link 1`] = `""`;
